src/schema.rs

```diff
@@ -50,10 +50,11 @@ impl Schema {
         self.types.get_type_by_name(name.as_ref())
     }
 
-    //TODO: change get_types() return type to SchemaTypeIterator and define SchemaTypeIterator
     /// Returns an iterator over the types in this schema.
-    pub(crate) fn get_types(&self) -> &HashMap<String, Type> {
-        todo!()
+    pub(crate) fn get_types(&self) -> SchemaTypeIterator {
+        SchemaTypeIterator {
+            types: self.types.get_types().iter(),
```

Comment:
I think this is ok for now since this isn't the final implementation, but I'm realizing that this will iterate over _all_ the types in the typestore. Since this is a method on `Schema`, this should only expose the types that are part of this schema.

Reply:
Reference issue: https://github.com/amzn/ion-schema-rust/issues/22